### DISCUSSION TITLE
Why does _close_event_loop close a closed loop?

### DISCUSSION THREAD

**Discussion by truh** [pytest-dev/pytest-asyncio#1006] [General]
```python
def _close_event_loop() -> None:
    policy = asyncio.get_event_loop_policy()
    try:
        loop = policy.get_event_loop()
    except RuntimeError:
        loop = None
    if loop is not None:
        if not loop.is_closed():
            warnings.warn(
                _UNCLOSED_EVENT_LOOP_WARNING % loop,
                DeprecationWarning,
            )
        loop.close()
```

There is a check right there to check whether the loop is closed or not, but than the loop is closed no matter if it's already closed or not.

Didn't want to open a bug because I might be missing something.

**Comment by seifertm** [comment]
The check was introduced much later than the call to _loop.close()_. The close method [is documented to be idempotent](https://docs.python.org/3/library/asyncio-eventloop.html#asyncio.loop.close), so closing a closed loop should have no effect.

Does the code cause any problem or do you think it should be improved otherwise?

**Reply by truh** [reply]
I got "Event loop is closed" runtime errors. But I guess this must come from trying to run something on the loop, not from repeated `.close()` calls.

I got the _UNCLOSED_EVENT_LOOP_WARNING in the same test and dug down to find this piece of code so I assumed this must be related. I have since fixed the underlying problems in my test case. Sorry to bother you.
